Completed Lesson-8

borisshavrin opened the pull request.

borisshavrin committed:
Completed Lesson-8

pantonov-gb commented on task1.py:
```diff
@@ -0,0 +1,38 @@
+            raise ValueError("Дней в месяце может быть от 1 до 31")
+        if month not in range(1, 13):
+            raise ValueError("Месяцев в году может быть от 1 до 12")
+        if year not in range(2021):
```
Чтобы узнать, что `year` находится в последовательности, нужно просмотреть все члены этой последовательности. Эффективнее будет `0 < year < 2021`

pantonov-gb commented on task1.py:
```diff
@@ -0,0 +1,38 @@
+
+class Data:
+    def __init__(self, date):
+        Data.date_str = date
```
Это не очень хорошая затея, установить атрибут класса здесь вот таким образом. Если мы начнем создавать объекты типа `Data`, каждый новый объект будет изменять атрибут у класса Data, таким образом польза от смены атрибута класса выглядит сомнительной. Обычно передранные в конструктор аргументы сохраняют в атрибуты объекта - `self.date_str = date`

pantonov-gb commented on task2.py:
```diff
@@ -0,0 +1,23 @@
+    делителя программа должна корректно обработать эту ситуацию и не завершиться с ошибкой."""
+
+
+class Check(Exception):
```
Что же, наверное "класс-исключение, который контролирует деление на ноль" действительно должен выглядеть так. Но нам будет полезнее пример собственного исключения из методички!

pantonov-gb reviewed (approved):
Ок.

Есть небольшие недочеты, но не будем затягивать.

borisshavrin committed:
Modified task1

Чтобы узнать, что year находится в последовательности, нужно просмотреть все члены этой последовательности. Эффективнее будет 0 < year < 2021

Обычно передранные в конструктор аргументы сохраняют в атрибуты объекта - self.date_str = date

borisshavrin committed:
Modified

1) исправлено print('----...') на print('-'*n)
2) export_product вызывается по объекту класса, а не самому классу

borisshavrin committed:
Modified task5

1) исправлено print('----...') на print('-'*n)
2) export_product вызывается по объекту класса, а не самому классу

borisshavrin committed:
Merge remote-tracking branch 'origin/Lesson-8' into Lesson-8
